### ISSUE TITLE
NetworkPolicy doesn't block traffic

### ISSUE THREAD

**Issue by floriankammermann** [projectcalico/canal#78]
## Expected Behavior
NetworkPolicy blocks traffic according to the documentation of kubernetes https://kubernetes.io/docs/concepts/services-networking/networkpolicies/

## Current Behavior
The network traffic is not blocked.

## Steps to Reproduce (for bugs)
The reproducer is in this repo https://github.com/floriankammermann/kubernetes-networkpolicy

## Context
I have to isolate pods from each other on L3. Kinda multitenancy.

## Your Environment
OS: Ubuntu Xenial
Kernel (uname -a): Linux kube2 4.4.0-78-generic #99-Ubuntu SMP Thu Apr 27 15:29:09 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
Install tools:
installed docker according to https://docs.docker.com/engine/installation/linux/ubuntu/ docker -v Docker version 17.03.1-ce, build c6d412e
installed kubernetes according to https://kubernetes.io/docs/setup/independent/install-kubeadm/
Installed canal according to https://github.com/projectcalico/canal/blob/master/k8s-install/README.md (version 1.6)

Kubernetes version (use kubectl version):
```
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", |swisscom@kube2:~$ 
GitTreeState:"clean", BuildDate:"2017-05-19T18:44:27Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}           |swisscom@kube2:~$ 
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", |swisscom@kube2:~$ 
GitTreeState:"clean", BuildDate:"2017-05-19T18:33:17Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by tmjd** [comment]
Thanks for the awesome issue submission.  This is one of the nicest and complete I've seen. :+1: 
I created https://github.com/floriankammermann/kubernetes-networkpolicy/pull/1 with what fixes the NetworkPolicy.  I can't totally explain things because it seems like the solution is a NetworkPolicy that does not match the structure laid out https://kubernetes.io/docs/api-reference/extensions/v1beta1/definitions/#_v1beta1_networkpolicyingressrule

Basically the what you had was opening up that port for everyone, and the port needed to be a sub-element to the from clause.

**Comment by floriankammermann** [comment]
Thanks just tested the pull request https://github.com/floriankammermann/kubernetes-networkpolicy/pull/1/commits/b65a781a6bc11ed2926ed66c38722cc2749c2128 and it works now, thanks alot
